### DISCUSSION TITLE
Upgrade bare metal install?

### DISCUSSION THREAD

**Discussion by rice1204** [zwave-js/zwave-js-ui#3285] [Q&A] [answered]
Pardon the very basic question, but I've been trying to figure this out and I'm stumped!

What's the best way to upgrade an existing bare metal install? I've only ever used the dockerized version before, but in a remote location, I have a bare metal setup I'd like to update. Can I simply clone and reinstall while preserving ./store?

> git clone https://github.com/zwave-js/zwave-js-ui
cd zwave-js-ui
yarn install
yarn run build
yarn start`

**Comment by kpine** [comment]
Z-Wave JS UI distributes [binaries](https://github.com/zwave-js/zwave-js-ui/releases/tag/v8.25.1) (under Assets) for multiple platforms (but not all). Are you able to use those? Then it's simply a matter of downloading the new file.

**Reply by rice1204** [reply]
Good thought. I wish! 

This particular remote device was setup on an RPI3 a long while back because that's the only thing I had at the time. Binaries are a no go because it needs armv6

**Reply by robertsLando** [reply, marked as answer]
To upgrade just pull, no need to clone:

```
git pull
yarn install
yarn build
yarn start
```

1. Pull latest changes from master
2. Install upgraded dependencies (if any)
3. Build frontend/backend files
4. Start application

**Comment by rice1204** [comment]
Ugh, I knew it was basic. I should have known. Git pull and rebuild worked beautifully.

Thanks @robertsLando !!
